src/NATS.Client.Core/Internal/SslStreamConnection.cs

```diff
@@ -30,39 +35,57 @@ public async ValueTask DisposeAsync()
         {
             try
             {
-#if NET6_0
-                _closeCts.Cancel();
-#else
+#if NET8_0_OR_GREATER
                 await _closeCts.CancelAsync().ConfigureAwait(false);
+#else
+                _closeCts.Cancel();
 #endif
                 _waitForClosedSource.TrySetCanceled();
             }
             catch
             {
             }
 
+#if NETSTANDARD2_0
+            _sslStream.Dispose();
+#else
             await _sslStream.DisposeAsync().ConfigureAwait(false);
+#endif
         }
     }
 
     [MethodImpl(MethodImplOptions.AggressiveInlining)]
     public async ValueTask<int> SendAsync(ReadOnlyMemory<byte> buffer)
     {
+#if NETSTANDARD2_0
+        MemoryMarshal.TryGetArray(buffer, out var segment);
+        await _sslStream.WriteAsync(segment.Array, segment.Offset, segment.Count, _closeCts.Token).ConfigureAwait(false);
+#else
```

Comment:
I think we really need to access the underlying array here, otherwise there is no fallback. But you're right, we shouldn't ignore the failure. An exception is the only thing I can think of. Do you think we can do something else to be more helpful?

Reply:
done